### PR TITLE
Refactor tests (using jest), Use QDL tslint config, Use QDL quest schema

### LINKED ISSUE #98: Local unit testing seems broken
I've had tests fail (correctly) in TravisCI that weren't caught by testing locally. Something seems wrong here.

### PR DESCRIPTION
**Pull request by smartin015**
There's a lot going on in here:

- First off, we're now inheriting from expedition-qdl's tslint config, and these files were updated to be style-consistent with the core repo. In particular there was a bunch of bad style in our tests and a lot of unused imports.
- Dependency injection was added for the "Mail" function because the handler tests were failing due to mail not existing. This lets us test whether/how mail is sent as a by-product of the handlers.
- Sequelize schema creation is now done by parsing an instance of the `Quest` Schema class given by expedition-qdl repo. Same goes for validation.
- Because a nullable date isn't treated well by reflect.metadata, the quest schema uses a `PLACEHOLDER_DATE` constant as a default when the field is not set. This must be removed before inserting into postgres, which is done with the `prepare()` function in Schema.ts.
- Some of our async tests were throwing exceptions, causing the tests to fail via timeout. I added some `catch()` logic and switched express handlers to `end()` instead of `send()` to make things end more quickly.

Fixes #98 
#65

### CODE REVIEW

File: app/models/Quests.test.ts [hunk 1]
```diff
@@ -58,56 +59,61 @@ describe('quest', () => {
...
       .then(() => {
-        return q.model.create(insertedQuest);
+        return q.create(insertedQuest);
       })
       .then(() => {
-        return q.model.create(expansionQuest);
+        return q.create(expansionQuest);
       })
-      .then(() => {done();})
+      .then(() => {console.log('Beforeach finished'); done();})
```
Comment (toddmedema) [inline app/models/Quests.test.ts]: Rm console logs? Multiple in this file, shouldn't need them if tests are passing?

File: app/models/Users.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import Config from '../config'
 import * as Sequelize from 'sequelize'
 import * as Bluebird from 'bluebird'
-import {AnalyticsEvent, AnalyticsEventAttributes, AnalyticsEventInstance} from './AnalyticsEvents'
+import {AnalyticsEvent} from './AnalyticsEvents'
 const Mailchimp = require('mailchimp-api-v3');
 const mailchimp = (Config.get('NODE_ENV') !== 'dev' && Config.get('MAILCHIMP_KEY')) ? new Mailchimp(Config.get('MAILCHIMP_KEY')) : null;
 
+const sequelize = require('sequelize');
```
Comment (toddmedema) [inline app/models/Users.ts]: Shouldn't this be capitalized?

File: package.json [hunk 3]
```diff
@@ -102,19 +105,51 @@
...
     "npm-check-updates": "^2.10.3",
     "sinon-express-mock": "^2.0.0",
+    "ts-jest": "^22.4.4",
     "webpack-dashboard": "^0.3.0",
     "webpack-dev-server": "^2.4.2"
   },
   "engines": {
     "node": "8.1.4"
   },
+  "jest": {
```
Comment (toddmedema) [inline package.json]: Seems odd to have this as part of package.json... is there a separate jest config file option?

File: preprocessor.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const tsc = require('typescript');
```
Comment (toddmedema) [inline preprocessor.js]: What's this file doing? Not seeing it referenced anywhere else in the diff

File: package.json (hunk 3 above)
Comment (smartin015) [reply package.json]: Sure - split it out into jestconfig.json

File: preprocessor.js (hunk 4 above)
Comment (smartin015) [reply preprocessor.js]: Ah yep... that was an old attempt to get jest to work. The ts-jest module apparently works without this. Removed.

**Pull request merged**